### ISSUE TITLE
DDIM Sampler on Stable-Diffusion do not work well with CFG guidance scale large than 6~7

### ISSUE THREAD

**Issue by Randolph-zeng** [huggingface/diffusers#1602]
### Describe the bug

I have noticed that the SD v1.4 or v1.5 works poorly if I swap the scheduler to DDIM and have guidance scale larger than 7.
This behavior does not seems to be obvious on other sampler such as the default PNDM scheduler though. At first I suspect it is related to the "train-test-mismatch" mentioned in Imagen paper Sec 2.3.  However, **I found that that the same DDIM sampler in WebUI does not suffer from the same performance degradation with the same guidance scale.** I have manually traced the scales of the predicted epsilon value under the same prompt/guidance scale in both WebUI DDIM sampler and diffuser DDIM sampler, they are all within simialr range of [-4, 4]. I have also printed out the beta schedules of both but they are very very close( of course **I tried replacing diffusers DDIM betas with WebUI's DDIM betas but it does not help** )
<img width="557" alt="Screenshot 2022-12-08 at 15 08 04" src="https://user-images.githubusercontent.com/11933185/206381558-8c4ff9a1-dbd4-4073-bcf6-a3d62d8ced76.png">

### Reproduction

Reproduce is easy, and the behavior of following code snippet is consistent across diffusers versions from 0.3 to 0.9

```
from diffusers import StableDiffusionPipeline, DDIMScheduler
# swap any SD model here
pipe = StableDiffusionPipeline.from_pretrained('/xxxx/stable-diffusion-v1-5').to(0)
pipe.scheduler = DDIMScheduler.from_config(pipe.scheduler.config)
prompt = "A photograph of an astronaut riding a horse on Mars, high resolution, high definition."
# change the guidance scale from 1 - 15 and observe the performance degradation
image = pipe(prompt=prompt, num_inference_steps=100, guidance_scale=10., strength=1.).images[0]
```
guidance_scale = 2.5
<img width="510" alt="Screenshot 2022-12-08 at 15 19 25" src="https://user-images.githubusercontent.com/11933185/206383653-4d8e4a86-ef97-4676-8d8f-6efa5d3d2214.png">

guidance_scale = 5.0
<img width="509" alt="Screenshot 2022-12-08 at 15 20 10" src="https://user-images.githubusercontent.com/11933185/206383789-c4b13e05-aff9-4244-a0ac-5b775a64f4e2.png">

guidance_scale = 7.5
<img width="513" alt="Screenshot 2022-12-08 at 15 22 16" src="https://user-images.githubusercontent.com/11933185/206384175-7cb5959f-9001-40f4-a645-937312c0cfd2.png">

guidance_scale = 10.
<img width="513" alt="Screenshot 2022-12-08 at 15 21 10" src="https://user-images.githubusercontent.com/11933185/206383953-ce7d1b10-511c-4a73-bacc-1b683933f5f7.png">

### System Info

Both diffusers == 0.3.0 and diffusers == 0.9.0 suffer from such issue
Also SD v1-4 and v1-5 suffer from such issue

**Comment by anton-l** [comment]
cc @patrickvonplaten @patil-suraj this could be related to the missing dynamic thresholding

**Comment by Randolph-zeng** [comment]
@anton-l @patrickvonplaten @patil-suraj  Hi thanks for the response! However I don't think this is related to the missing dynamic thresholding. I did the same thing using the same model, same prompt, same sampler in **webUI(where DDIM functions normally there)**. I used a debugger to trace through the code and **I believe there is no dynamic thresholding there**. In fact they are using the sampler implemented in the runwayml/stable-diffusion repo. I suspect it is the formula but I don't see any inconsistence and beta calculation is very close
